pkg/kv/kvserver/kvserverbase/forced_error.go

```diff
@@ -72,34 +72,51 @@ func CheckForcedErr(
 	raftCmd *kvserverpb.RaftCommand,
 	isLocal bool,
 	replicaState *kvserverpb.ReplicaState,
-) ForcedErrResult {
+) (res ForcedErrResult) {
+	isLeaseRequest := raftCmd.ReplicatedEvalResult.IsLeaseRequest
+	var requestedLease roachpb.Lease
+	if isLeaseRequest {
+		requestedLease = *raftCmd.ReplicatedEvalResult.State.Lease
+	}
+
+	defer func() {
+		if res.ForcedError != nil && isLeaseRequest {
+			log.Infof(ctx,
+				"rejected lease request %s; current lease %s; err: %s",
+				requestedLease, *replicaState.Lease, res.ForcedError)
+		}
+		if raftCmd.ReplicatedEvalResult.IsLeaseTransfer {
```

Comment:
It looks like this isn't conditional on a non-nil error but was intended to be.  Perhaps we can early return for a nil error at the beginning of the defer?

Reply:
Yep can confirm that this logs spuriously.

> I241114 11:46:35.636960 2226 kv/kvserver/kvserverbase/forced_error.go:90 ⋮ [T1,Vsystem,n3,s3,r3540/3:‹/Table/106/1/-418{4504…-3889…}›,raft] 14476  rejected lease transfer repl=(n2,s2):2 seq=7 start=1731584795.531805573,0 exp=1731584801.528864866,0 pro=1731584795.528864866,0 acq=Transfer; current lease repl=(n3,s3):3 seq=6 start=1731582944.484704600,0 epo=4 min-exp=0,0 pro=1731582944.484704600,0 acq=Request; err: <nil>

Reply:
Yup, it was. Thanks for catching -- fixed.
<!-- Reviewable comment -OBftoovDnOT-ye6KPcA:bk2mk2u -->
<!-- Sent from Reviewable.io -->